COMPILER_ARCHITECTURE.md

```diff
@@ -0,0 +1,1430 @@
+   - Options: TypeScript, JavaScript, WASM
+
+2. **Standalone Executable**
+   - Native performance
+   - No runtime dependencies
+   - Options: LLVM, Rust, C, Zig, Cranelift
+
+3. **Nice-to-have: Microcontrollers**
+   - Embedded systems
+   - Resource-constrained environments
+
+### Browser Targets
+
+#### Option 1: WebAssembly (WASM)
+
+**Pros**:
+- Near-native performance
+- Compact binary format
+- Security sandbox
+- Well-supported
+
+**Cons**:
+- Requires runtime (WASM engine)
+- FFI with JS can be complex
+- Debugging harder than JS
+
+**Approach**: Use `wasm32-unknown-unknown` target, compile IR to WASM bytecode.
+
+```rust
+pub struct WasmBackend {
+    module: wasm_encoder::Module,
+}
+
+impl WasmBackend {
+    fn compile_function(&mut self, func: &MonomorphicFunction) {
+        // Emit WASM instructions
+    }
+}
+```
+
+#### Option 2: TypeScript
+
+**Pros**:
+- Type safety
+- Good tooling
+- Easy debugging
+- Source maps
+
+**Cons**:
+- Requires transpilation step
+- Larger output than WASM
+- TypeScript runtime needed
+
+**Approach**: Emit TypeScript code from IR, use `tsc` or `esbuild` for bundling.
+
+#### Option 3: JavaScript
+
+**Pros**:
+- No compilation needed
+- Universal browser support
+- Easy debugging
+
+**Cons**:
+- No static type checking
+- Larger output
+- Less efficient than WASM
+
+**Approach**: Emit JS code directly from IR.
+
+#### Recommendation for Browser
+
+**Hybrid approach**:
+- **AOT (Ahead-of-time)**: Compile to WASM for production
+  - Best performance
+  - Smallest bundle size
+  
+- **In-browser**: Compile to JavaScript for development
+  - Easier debugging
+  - Source maps
+  - No WASM compilation complexity
+
+The compiler can target both from the same IR.
+
+### Standalone Targets
+
+#### Option 1: LLVM
+
+**Pros**:
+- Industry-standard
+- Excellent optimizations
+- Many target architectures
+- Used by Rust, Swift, etc.
+
+**Cons**:
+- Large dependency
+- Complex API
+- Longer compile times
+
+**Approach**: Use `inkwell` crate for LLVM bindings.
+
+```rust
+pub struct LlvmBackend<'ctx> {
+    context: &'ctx Context,
+    module: Module<'ctx>,
+    builder: Builder<'ctx>,
+}
+```
+
+#### Option 2: Emit Rust
+
+**Pros**:
+- Leverage Rust compiler
+- Excellent error messages
+- Already set up in our workflow
+- Good debugging
+
+**Cons**:
+- Requires Rust toolchain
+- Slower compilation
+- Not truly standalone
+
+**Approach**: Emit Rust code from IR, invoke `rustc`.
+
+#### Option 3: Emit C
+
+**Pros**:
+- Universal
+- Highly portable
+- Mature toolchain
+- Works everywhere
+
+**Cons**:
+- Manual memory management
+- Less type safety
+- Older language
+
+**Approach**: Emit C code from IR, invoke `gcc` or `clang`.
+
+#### Option 4: Cranelift
+
+**Pros**:
+- Fast compilation
+- Designed for JIT/AOT
+- Used by Wasmtime
+- Pure Rust
+
+**Cons**:
+- Less optimization than LLVM
+- Fewer target architectures
+- Younger project
+
+**Approach**: Use Cranelift as a library for code generation.
```

Comment:
I dunno. I think this has the same downsides as wasm. Lot's over work required for lowering. Targeting rust would be probably the least amount of work on the backend, from my perspective.